lib/docker-toolbelt.coffee

```diff
@@ -107,12 +112,49 @@ EEXIST = code: 'EEXIST'
 ignore = ->
 MIN_PAGE_SIZE = 4096
 
+pathPrefixRemover = (prefix) ->
+	(path) ->
+		slice = path.substr(prefix.length)
+		# return original if path doesn't start with given prefix
+		return if "#{prefix}#{slice}" == path then slice else path
+
+overlay2MountWithDisposer = (fsRoot, target, lowers, diffDir, workDir) ->
+	# If no lower, just return diff directory
+	return Promise.resolve(diffDir) if !lowers
+
+	fs.mkdirAsync(target)
+	.catch EEXIST, ignore
+	.then ->
+		options = "lowerdir=#{lowers},upperdir=#{diffDir},workdir=#{workDir}"
+		mountOpts = {}
+		# Use relative paths when the mount data has exceeded the page size.
+		# The mount syscall fails if the mount data cannot fit within a page and
+		# relative links make the mount data much smaller.
+		if options.length > MIN_PAGE_SIZE
+			mountOpts.cwd = fsRoot
+			makeRelative = pathPrefixRemover(fsRoot + '/')
+			options = [
+				"lowerdir=#{lowers.split(':').map(makeRelative).join(':')}"
+				"upperdir=#{makeRelative(diffDir)}"
+				"workdir=#{makeRelative(workDir)}"
+			].join(',')
+		execAsync("mount -t overlay -o '#{options}' none #{target}", mountOpts)
+	.return(target)
```

Comment:
Too tired to test now but I think `.return` is needed there to nudge Bluebird to resolve the promise with this value, otherwise it'll resolve with the result of `execAsync` above. I'm using the resolved value in the disposer just for consistency with how disposers are used in general and while it's true it's not strictly required, I prefer that it's passed as an argument.